File: Data Structures & Algorithms/non-cyclical-number/submission-3.js
class Solution {
    /**
     * @param {number} n
     * @return {boolean}
     */
    getDigit(n) {
        var res = 0;
        while(n > 0) {
            res += Math.pow(n % 10, 2);
            n = Math.floor(n / 10);
        }
        return res;
    }

    isHappy(n) {
        var tracker = new Map();
        while(n > 1) {
            n = this.getDigit(n);
            if(tracker.has(n))
                return false;
            tracker.set(n, 1);
        }
        return true;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-3.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        Dictionary<char, int> tracker = new Dictionary<char, int>();
        for (int i = 0; i < s.Length; i++) {
            int d = 1;
            if (tracker.ContainsKey(s[i])) {
                d += tracker[s[i]];
            }
            tracker[s[i]] = d;
        }

        for (int i = 0; i < t.Length; i++) {
            if(!tracker.ContainsKey(t[i]) || tracker[t[i]] < 0)
                return false;
            tracker[t[i]]--;
        }
        return true;
    }
}
